File: projects/order-creation-queries/core/src/main/java/com/robinfood/core/mappers/OrderDailyMappers.java
package com.robinfood.core.mappers;

import com.robinfood.core.dtos.OrderDailyDTO;
import com.robinfood.core.entities.OrderDailyEntity;

import java.util.List;
import java.util.stream.Collectors;

public final class OrderDailyMappers {

    private OrderDailyMappers() {
        throw new IllegalStateException("Utility class");
    }

    public static List<OrderDailyDTO> orderDailyEntityToOrderDailyDtoList(List<OrderDailyEntity> orderDailyEntityList) {
        return orderDailyEntityList.stream().map(
                OrderDailyMappers::orderDailyEntityToOrderDailyDto
        ).collect(Collectors.toList());
    }

    public static OrderDailyDTO orderDailyEntityToOrderDailyDto(
            OrderDailyEntity orderDailyEntity
    ) {
        return OrderDailyDTO.builder()
                .brandId(orderDailyEntity.getBrandId())
                .createdAt(orderDailyEntity.getCreatedAt())
                .deliveryTypeId(orderDailyEntity.getDeliveryTypeId())
                .id(orderDailyEntity.getId())
                .orderInvoiceNumber(orderDailyEntity.getOrderInvoiceNumber())
                .orderNumber(orderDailyEntity.getOrderNumber())
                .origin(OriginMappers.originEntityToOriginDto(orderDailyEntity.getOrigin()))
                .total(orderDailyEntity.getTotal())
                .user(OrderDetailUserMappers.toOrderDetailUserMappers(orderDailyEntity.getUser()))
                .build();
    }

}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/mappers/OrderDailyMappers.kt
package com.robinfood.core.mappers

import com.robinfood.core.dtos.OrderDailyDTO
import com.robinfood.core.entities.orderdaily.OrderDailyEntity
import java.math.BigDecimal

fun OrderDailyEntity.toOrderDailyDTO(): OrderDailyDTO?{
    return if (id == null) {
        null
    } else {
        return OrderDailyDTO(
            id = id,
            orderNumber = orderNumber.orEmpty(),
            invoiceNumber = orderInvoiceNumber.orEmpty(),
            totalPrice = total ?: BigDecimal.ZERO,
            deliveryTypeId = deliveryTypeId,
            originId = origin.id,
            origin = origin.name,
            originBackgroundColor = origin.color,
            brandId = brandId,
            brandImage = brand.image,
            brandBackgroundColor = brand.color,
            userId = user.id,
            customerName = user.firstName +" "+ user.lastName,
            creationTime = createdAt
        )
    }
}